js/core.js

```diff
@@ -1723,8 +1726,16 @@
 
             //This check has been added to show "null" in all the rows if the user inputs blank string
             //We are opting json out here because we want null in the UI instead of "", so we do not call _getNullValue for json
-            if (data === undefined || (data === null && this.type.name.indexOf('json') !== 0)) {
+            if (data === undefined || (data === null && this.type.name.indexOf('json') === -1)) {
                 return this._getNullValue(context);
+            } else if (data === null && this.type.name.indexOf('json') !== 0) {
+                return data;
+            }
+
+            var display = this.getDisplay(context);
+
+            if (display.isPreformat) {
+                return module._printf(display.preformatConfig, data);
```

Comment:
This function might throw an error which we don't want to raise to the users. Wrap this in a try-catch to ignore the pre_format annotation in that case.